src/server/pegasus_server_impl.cpp

```diff
@@ -2558,6 +2558,27 @@ void pegasus_server_impl::update_rocksdb_iteration_threshold(
     }
 }
 
+void pegasus_server_impl::update_rocksdb_block_cache_enabled(
+    const std::map<std::string, std::string> &envs)
+{
+    bool cache_enabled = _data_cf_rd_opts.fill_cache;
+    auto find = envs.find(ROCKSDB_BLOCK_CACHE_ENABLED);
+    if (find != envs.end()) {
+        if (!dsn::buf2bool(find->second, cache_enabled)) {
+            derror_replica("{}={} is invalid.", find->first, find->second);
+            return;
+        }
+    }
+
```

Comment:
If this `ROCKSDB_BLOCK_CACHE_ENABLED` is not found in app_envs,  `cache_enabled` should be set as default value, but in your implementation, this value won't be updated.
For example, this default value is `true`. Firstly, admin user update it as `false` through adding an envs. Then **this app_envs is deleted**, the value is expected to be `true`, not `false`.